com.unity.shadergraph/Editor/Generation/Processors/CustomInterpolatorUtils.cs

```diff
@@ -50,22 +50,25 @@ internal static class Splice
         [GenerationAPI]
         internal struct Descriptor
         {
-            internal string src, dst; // for function or block.
+            internal string src, dst; // for function or block. For macro block src is start of the macro and dst is end of the macro.
             internal string name;     // for struct or function.
             internal string define;   // defined for client code to indicate we're live.
             internal string splice;   // splice location, prefer use something from the list.
             internal string preprocessor;
+            internal bool hasMacro;
 
-            internal bool isBlock => src != null && dst != null && name == null && splice != null;
+            internal bool isBlock => src != null && dst != null && name == null && splice != null && !hasMacro;
+            internal bool isMacroBlock => src != null && dst != null && name == null && splice != null && hasMacro;
             internal bool isStruct => src == null && dst == null && name != null && splice != null;
             internal bool isFunc => src != null && dst != null && name != null && splice != null;
             internal bool isDefine => define != null && splice != null && src == null && dst == null & name == null;
-            internal bool isValid => isDefine || isBlock || isStruct || isFunc;
+            internal bool isValid => isDefine || isBlock || isStruct || isFunc || isMacroBlock;
             internal bool hasPreprocessor => !String.IsNullOrEmpty(preprocessor);
 
             internal static Descriptor MakeFunc(string splice, string name, string dstType, string srcType, string define = "", string preprocessor = "") => new Descriptor { splice = splice, name = name, dst = dstType, src = srcType, define = define, preprocessor = preprocessor };
             internal static Descriptor MakeStruct(string splice, string name, string define = "", string preprocessor = "") => new Descriptor { splice = splice, name = name, define = define, preprocessor = preprocessor };
             internal static Descriptor MakeBlock(string splice, string dst, string src, string preprocessor = "") => new Descriptor { splice = splice, dst = dst, src = src, preprocessor = preprocessor };
+            internal static Descriptor MakeMacroBlock(string splice, string startMacro, string endMacro, string preprocessor = "") => new Descriptor { splice = splice, dst = endMacro, src = startMacro, preprocessor = preprocessor, hasMacro = true };
```

Comment:
This doesn't seem to be used in the PR?

Reply:
It is used in HD Targets, this is use for the Macro to Interpolate in the domain shader:
CustomInterpSubGen.Descriptor.MakeMacroBlock("CustomInterpolatorInterpolateWithBaryCoordsMeshToDS", "TESSELLATION_INTERPOLATE_BARY(", ", baryCoords)"),